Sources/SwiftFormat/PrettyPrint/TokenStreamCreator.swift

```diff
@@ -2716,18 +2732,40 @@ fileprivate final class TokenStreamCreator: SyntaxVisitor {
     extractLeadingTrivia(token)
     closeScopeTokens.forEach(appendToken)
 
+    generateEnable(Selection.Range(
+      start: token.positionAfterSkippingLeadingTrivia,
+      end: token.endPositionBeforeTrailingTrivia))
+
     if !ignoredTokens.contains(token) {
       // Otherwise, it's just a regular token, so add the text as-is.
       appendToken(.syntax(token.presence == .present ? token.text : ""))
     }
 
+    generateDisable(token.endPositionBeforeTrailingTrivia)
+
     appendTrailingTrivia(token)
     appendAfterTokensAndTrailingComments(token)
 
     // It doesn't matter what we return here, tokens do not have children.
     return .skipChildren
   }
 
+  func generateEnable(_ range: Selection.Range) {
```

Comment:
I would name this something like `generateEnableFormattingIfNecessary`. `generateEnable` to me sounds like it unconditionally enables formatting. Same for `generateDisable`.

---

Can `generateEnable` and `generateDisable` be `private`?